_gsocproposals/2019/proposal_WLCGansible.md

```diff
@@ -0,0 +1,57 @@
+
+The SIMPLE Grid project is an extension of the SIMPLE Framework that combines popular configuration management technologies such as Puppet/Ansible and container orchestration technologies such as Docker Swarm/Kubernetes to allow deployment of complex computing clusters using a single site level configuration file. The various components of the framework and their functions are described in the SIMPLE specification document. Two of the core components written in Python are: SIMPLE Grid YAML Compiler and SIMPLE Grid Validation Engine.   
+
+
+
+## Task ideas and expected results
+
+Through this project, in broad terms, the student participant can expect to extend and improve the Central Configuration Manager implemented in Ansible during Google Summer of Code 2018. 
+The functions of the Central Configuration Manager are described in the SIMPLE framework’s specification document. They have already been implemented in the simple_grid_puppet_module available on Github. Using these resources, the student can expect to work towards the following deliverables:
+
+* Create roles for Config_Master (CM) and Lightweight_Component (LC) nodes: The framework abstracts the computing infrastructure into the above-mentioned node types. 
+* Implement Ansible playbooks for Installation stage : Ansible tasks related to configuring Ansible, downloading the Ansible module for the SIMPLE Grid framework
+* Implement Ansible playbooks for Configuration stage: Ansible tasks to configure the SIMPLE Grid Ansible module. Compile the site level configuration file via the SIMPLE Grid YAML compiler and validate its contents using the Simple Grid Validation Engine
+* Implement Ansible playbooks for Pre-Deployment stage: Set up a container orchestrator. In case of Docker Swarm, set up an attachable overlay network and a DNS file. Copy the output of the YAML compiler to all the Lightweight Component nodes and copy lifecycle callbacks for component repositories to appropriate nodes based on the execution_id generated by the compiler.
+* Implement Ansible playbooks for Deployment stage: Configure Firewall, CVMFS on lightweight component nodes, execute lifecycle hooks and lifecycle events of component repositories and start the Docker containers for each component repository.
+* Implement logging infrastructure for each stage.
+* Implement tracking of various deployment stages.
+
+
+
+**Requirements**
+
+* Familiarity and prior experience with Ansible.
+* Programming experience in Python is a plus.
+* Linux string fundamentals.
+* Git/Agile development methodology.
+
+
+
+**Mentors**
```

Comment:
Please add the mentors with their affiliations to gsoc/2019/mentors.md

Reply:
done